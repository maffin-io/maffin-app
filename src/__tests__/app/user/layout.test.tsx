import React from 'react';
import { render, waitFor } from '@testing-library/react';
import * as swr from 'swr';

import AccountLayout from '@/app/user/layout';

describe('AccountLayout', () => {
  beforeEach(() => {
    swr.mutate('/state/theme', undefined);
  });

  it('renders as expected', async () => {
    const { container } = render(
      <AccountLayout>
        <span>child</span>
      </AccountLayout>,
    );
    const html = document.documentElement;
    await waitFor(() => expect(html).toHaveClass('dark'));

    expect(container).toMatchSnapshot();
  });

  it('sets localstorage theme', async () => {
    localStorage.setItem('theme', 'light');
    render(
      <AccountLayout>
        <span data-testid="child">child</span>
      </AccountLayout>,
    );

    const html = document.documentElement;
    await waitFor(() => expect(html.classList).toHaveLength(0));
  });
});
