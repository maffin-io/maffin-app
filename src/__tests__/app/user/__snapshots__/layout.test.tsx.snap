// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountLayout renders as expected 1`] = `
<div>
  <div
    class="flex h-screen place-content-center place-items-center"
  >
    <div
      class="card flex items-center"
    >
      <img
        alt=""
        class="m-0"
        data-nimg="1"
        decoding="async"
        height="54"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=128&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=64&q=75 1x, /_next/image?url=%2Fimg.jpg&w=128&q=75 2x"
        style="color: transparent;"
        width="54"
      />
      <div
        class="p-4"
      >
        <span>
          child
        </span>
      </div>
    </div>
  </div>
  <footer
    class="bottom-0 w-full flex justify-end bg-white dark:bg-dark-700 text-sm px-5 z-1"
  >
    2023
    Â©
     
    maffin.io
  </footer>
</div>
`;
